tsdb/index/inmem/meta.go

```diff
@@ -1455,12 +1455,26 @@ func (m *Measurement) TagKeys() []string {
 }
 
 // TagValues returns all the values for the given tag key, in an arbitrary order.
-func (m *Measurement) TagValues(key string) []string {
+func (m *Measurement) TagValues(auth query.Authorizer, key string) []string {
 	m.mu.RLock()
 	defer m.mu.RUnlock()
 	values := make([]string, 0, len(m.seriesByTagKeyValue[key]))
-	for v := range m.seriesByTagKeyValue[key] {
-		values = append(values, v)
+VALUES:
+	for v, series := range m.seriesByTagKeyValue[key] {
```

Comment:
Looks fine, but I think a sprinkling of documentation here wouldn't go amiss. Mainly along the lines of explaining that we're looking for a single series containing the tag value that we're authorised to read.